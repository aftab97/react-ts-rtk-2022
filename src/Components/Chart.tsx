import React, { useEffect, useState } from "react";

import moment from "moment";
import { AgeGraph } from "./AgeGraph";
import { YoeGraph } from "./YoeGraph";
import { SalaryGraph } from "./SalaryGraph";

const Chart = ({ users }: any) => {
  const [ageData, setAgeData] = useState<any[]>([]);
  const [yoeData, setYoeData] = useState<any[]>([]);
  const [salaryData, setSalaryData] = useState<any[]>([]);

  const testData = users.map(({ x, y }: any) => {
    return { x, y };
  });

  let allData = {
    graphAgeData: {
      datasets: [
        {
          label: "Age",
          data: users.map(({ x, age }: any) => {
            return { x, y: age };
          }),
          backgroundColor: "rgba(255, 99, 132, 1)",
        },
      ],
    },

    graphSalaryData: {
      datasets: [
        {
          label: "Salary",
          data: users.map(({ x, salary }: any) => {
            return { x, y: salary };
          }),
          backgroundColor: "blue",
        },
      ],
    },

    graphYoeData: {
      datasets: [
        {
          label: "Years Of Experience",
          data: users.map(({ x, years_of_experience }: any) => {
            return { x, y: years_of_experience };
          }),
          backgroundColor: "purple",
        },
      ],
    },
  };

  const allOptions = {
    ageDataOptions: {
      scales: {
        y: {
          beginAtZero: true,
        },
      },
    },
  };

  return (
    <div>
      <AgeGraph
        graphData={allData.graphAgeData}
        dataOptions={allOptions.ageDataOptions}
      />

      <YoeGraph
        graphData={allData.graphYoeData}
        dataOptions={allOptions.ageDataOptions}
      />

      <SalaryGraph
        graphData={allData.graphSalaryData}
        dataOptions={allOptions.ageDataOptions}
      />
    </div>
  );
};

export default Chart;
