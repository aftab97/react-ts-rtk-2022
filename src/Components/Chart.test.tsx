import React from "react";
import renderer from "react-test-renderer";
import { Chart } from "./Chart";

//when first created creates a file in the test dir creating a SS of the button component and will
//check to see if any changes occur and if they do not match then it fails
it("renders correctly", () => {
  const tree = renderer.create(<Chart />).toJSON();
  expect(tree).toMatchSnapshot();
});

// snapshots are used when someone makes change to your code without permission so that way its a fail safe from stopping
//unexpected code being merged into the master branch
