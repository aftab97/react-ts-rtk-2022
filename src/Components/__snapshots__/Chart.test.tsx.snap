// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>
  <div>
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>
  <div>
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>
</div>
`;
